src/main/kotlin/org/jitsi/nlj/transform/node/incoming/RtcpTermination.kt

```diff
@@ -79,16 +78,21 @@ class RtcpTermination(
             packetReceiveCounts.merge(rtcpPacket::class.simpleName!!, 1, Int::plus)
             rtcpEventNotifier.notifyRtcpReceived(rtcpPacket, packetInfo.receivedTime)
 
-            if (rtcpPacket is RtcpSrPacket) {
+            (forwardedRtcp as? RtcpSrPacket)?.let {
                 // NOTE(george) effectively eliminates any report blocks as we don't want to relay those
-                logger.cdebug { "saw an sr from ssrc=${rtcpPacket.senderSsrc}, timestamp=${rtcpPacket.senderInfo.rtpTimestamp}" }
-                val lengthBytes = RtcpHeader.SIZE_BYTES + SenderInfoParser.SIZE_BYTES
-                rtcpPacket.length = lengthBytes
-                // We can do this because we've already parsed the compound packet and we are discarding it anyway
-                rtcpPacket.lengthField = (lengthBytes / 4) - 1
+                logger.cdebug { "Saw an sr from ssrc=${rtcpPacket.senderSsrc}, timestamp=${it.senderInfo.rtpTimestamp}" }
+                forwardedRtcp = if (it.reportCount > 0) {
+                    it.cloneWithoutReportBlocks()
+                } else {
+                    it
+                }
             }
         }
         return if (forwardedRtcp != null) {
+            if (forwardedRtcp?.buffer != packetInfo.packet.buffer) {
```

Comment:
it's still needed since `fowardedRtcp` is var.  i could change it to:
```
        return forwardedRtcp?.let {
            if (it.buffer != packetInfo.packet.buffer) {
                // We're not using the original packet's buffer, so we can return it to the pool
                BufferPool.returnBuffer(packetInfo.packet.buffer)
            }
            // Manually cast to RtcpPacket as a workaround for https://youtrack.jetbrains.com/issue/KT-7186
            packetInfo.packet = forwardedRtcp as RtcpPacket
            packetInfo
        } ?: run {
            packetDiscarded(packetInfo)
            null
        }
```

Reply:
Whichever you think is clearer.  If you do this you should probably have `packetInfo.packet = it as RtcpPacket` tho.

Reply:
gah, yup. will fix.

Reply:
looks like we no longer need the cast with this way, too.